IDEAS/BoundaryConditions/SolarIrradiation/RadSolData.mo

```diff
@@ -34,11 +34,18 @@ model RadSolData "Selects or generates correct solar data for this surface"
   Modelica.Blocks.Interfaces.RealOutput Tenv "Environment temperature"
     annotation (Placement(transformation(extent={{96,-30},{116,-10}})));
 protected
-  final parameter Boolean solDataInBus=if sum( { if sum(abs({mod(incAndAziInBus[i,1],Modelica.Constants.pi*2),mod(incAndAziInBus[i,2],Modelica.Constants.pi*2)} -
-  {mod(inc,Modelica.Constants.pi*2),mod(azi,Modelica.Constants.pi*2)}))<0.01 then 1 else 0 for i in 1:numIncAndAziInBus})   ==1 then true else false
+  final parameter Integer numMatches=
+    sum( {if     IDEAS.Utilities.Math.Functions.isAngle(incAndAziInBus[i,1],inc)
+             and IDEAS.Utilities.Math.Functions.isAngle(incAndAziInBus[i,2],azi)
+          then 1
+          else 0 for i in 1:numIncAndAziInBus});
+  final parameter Boolean solDataInBus = numMatches==1
     "True if the {inc,azi} combination is found in incAndAziInBus" annotation(Evaluate=true);
-  final parameter Integer solDataIndex=sum( { if sum(abs({mod(incAndAziInBus[i,1],Modelica.Constants.pi*2),mod(incAndAziInBus[i,2],Modelica.Constants.pi*2)} -
-  {mod(inc,Modelica.Constants.pi*2),mod(azi,Modelica.Constants.pi*2)}))<0.01 then i else 0 for i in 1:numIncAndAziInBus})
+  final parameter Integer solDataIndex=
+    sum( {if     IDEAS.Utilities.Math.Functions.isAngle(incAndAziInBus[i,1],inc)
+             and IDEAS.Utilities.Math.Functions.isAngle(incAndAziInBus[i,2],azi)
+          then i
+          else 0 for i in 1:numIncAndAziInBus})
```

Comment:
The previous implementation allowed some numerical errors (it turned out it was necessary to have it). Why did you remove it?

Reply:
Ok, you have modified the isAngle.

Reply:
I moved the previous implementation to the function `isAngle` to have a better readable code.